server/routes/witnesses/addWitness.ts

```diff
@@ -0,0 +1,199 @@
+  ) {}
+
+  private helpers = new Map<WitnessAnswerType, DecisionHelper>([
+    [WitnessAnswerType.STAFF, new StaffDecisionHelper(this.userService, this.decisionTreeService)],
+    [WitnessAnswerType.OFFICER, new OfficerDecisionHelper(this.userService, this.decisionTreeService)],
+    [WitnessAnswerType.OTHER_PERSON, new OtherPersonWitnesDecisionHelper(this.decisionTreeService)],
+  ])
+
+  private renderView = async (req: Request, res: Response, pageData?: PageData): Promise<void> => {
+    const { adjudicationNumber, errors } = pageData
+    const { user } = res.locals
+    const prisoner = await this.placeOnReportService.getPrisonerDetailsFromAdjNumber(adjudicationNumber, user)
+    const selectedAnswerViewData = await this.answerTypeHelper(pageData)?.viewDataFromForm(pageData, user)
+
+    return res.render(`pages/addWitness`, {
+      errors: errors || [],
+      decisionForm: pageData,
+      selectedAnswerViewData,
+      pageData,
+      prisoner,
+      cancelButtonHref: adjudicationUrls.detailsOfWitnesses.urls.modified(adjudicationNumber),
+    })
+  }
+
+  view = async (req: Request, res: Response): Promise<void> => {
+    const adjudicationNumber = Number(req.params.adjudicationNumber)
+    const selectedAnswerId = req.query.selectedAnswerId as string
+    const selectedPerson = req.query.selectedPerson as string
+    if (selectedPerson && selectedAnswerId) {
+      // We are coming back from a user selection.
+      const answerTypeHelper = this.answerTypeHelper(selectedAnswerId)
+      const form = answerTypeHelper.formAfterSearch(selectedAnswerId, selectedPerson)
+
+      return this.renderView(req, res, { ...form, adjudicationNumber })
+    }
+    if (req.query.selectedAnswerId) {
+      // We are coming from delete.
+      return this.renderView(req, res, { adjudicationNumber, selectedAnswerId })
+    }
+    // We are viewing the page normally
+    return this.renderView(req, res, { adjudicationNumber })
+  }
+
+  submit = async (req: Request, res: Response): Promise<void> => {
+    const { user } = res.locals
+    const adjudicationNumber = Number(req.params.adjudicationNumber)
+    const { selectedAnswerId } = req.body
+    if (req.body.deleteUser) {
+      return this.deleteUser(req, res, adjudicationNumber)
+    }
+
+    if (req.body.searchUser) {
+      return this.search(req, res)
+    }
+    if (!selectedAnswerId) {
+      return this.renderView(req, res, { errors: [error.MISSING_DECISION], adjudicationNumber })
+    }
+    const answerTypeHelper = this.answerTypeHelper(selectedAnswerId)
+    const form = answerTypeHelper.formFromPost(req)
+    const errors = await answerTypeHelper.validateForm(form, req, res.locals.user)
+    if (errors && errors.length !== 0) {
+      return this.renderView(req, res, { errors, ...form, adjudicationNumber })
+    }
+
+    const witnessName = await answerTypeHelper.witnessNamesForSession(form, user)
+
+    const witnessToAdd = {
+      code: this.convertToWitnessCode(selectedAnswerId),
```

Comment:
this is where i convert the enums needed to reuse the helpers to the enums used for the witnesses endpoint